### PR TITLE
#167733819 Add validations to create request endpoint

### PR DESCRIPTION
**Pull request by codecell**
**What does this PR do?**
Adds validations to the 
POST /api/v1/request/ route created by @fegzycole 

**Description of task to be completed**
- create a validate request middleware
- implement the validation algorithms
- apply the validation middleware to the POST /api/v1/request/ route
- notify @evansinho  to add tests and swagger docs of this route

**How should this be manually tested?**
After this PR is merged, make a POST request to the endpoint /api/v1/request with any incomplete
data and and depending on the type of request (one-way, round-trip OR multi-city), you will get the requested error message , LOL.

**Any background context you want to provide?**
This adds  a validations middleware for the endpoint  created by @fegzycole 
@evansinho to create  tests and swagger doc for the endpoint

**What are the relevant pivotal tracker stories?**
#167733818
#167733819
#167733822

### CODE REVIEW

File: src/controllers/request.js [hunk 1]
```diff
@@ -17,14 +19,49 @@ export default class RequestController {
    */
   static async getAllRequests(req, res) {
     try {
-      const allRequests = await Request.findAll({
-        where: {
```
Comment (EziamakaNV) [inline src/controllers/request.js]: The where clause is intended to get all requests made by a PARTICULAR user to see all the requests that they have submitted over time to BareFoot Nomad

Corresponding PT Story: #167733825

Comment (Ileriayo) [reply src/controllers/request.js]: What action would you advice to be carried out? @EziamakaNV

**Review by fegzycole — COMMENTED** [review]
Great Job brother!
This validation is quite comprehensive.
However, I noticed that the checkToken middleware was commented out.

**Review by fegzycole — APPROVED** [review]
Looks good!

**Review by Oluwafayokemi — APPROVED** [review]

**Pull request merged**